WalletWasabi/Blockchain/TransactionOutputs/CoinsRegistry.cs

```diff
@@ -170,6 +182,17 @@ public void SwitchToUnconfirmFromBlock(Height blockHeight)
 				foreach (SmartCoin createdCoin in allCoins.CreatedBy(txId))
 				{
 					toRemove.AddRange(Remove(createdCoin));
+					foreach (var removedCoin in toRemove)
+					{
+						var removedCoinOutPoint = removedCoin.GetOutPoint();
+						if (TryGetSpenderSmartCoinByOutPoint(removedCoinOutPoint, out var coinByOutPoint))
```

Comment:
This is incorrect, but strangely it works. You are calling this function that tries to acquire the same lock you are in. However for some strange reason it succeeds to acquire the lock :o

Reply:
I fixed this.